src/main/java/org/opensearch/ad/rest/RestGetAnomalyDetectorAction.java

```diff
@@ -129,4 +143,35 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                 )
             );
     }
+
+    private Entity buildEntity(RestRequest request, String detectorId) throws IOException {
+        if (Strings.isEmpty(detectorId)) {
+            throw new IllegalStateException(CommonErrorMessages.AD_ID_MISSING_MSG);
+        }
+
+        String entityName = request.param(CommonName.CATEGORICAL_FIELD);
+        String entityValue = request.param(CommonName.ENTITY_KEY);
+
+        if (entityName != null && entityValue != null) {
+            // single-stream profile request:
+            // GET _opendistro/_anomaly_detection/detectors/<detectorId>/_profile/init_progress?category_field=<field-name>&entity=<value>
+            return Entity.createSingleAttributeEntity(detectorId, entityName, entityValue);
+        } else if (request.hasContent()) {
+            /* HCAD profile request:
+             * GET _opendistro/_anomaly_detection/detectors/<detectorId>/_profile/init_progress
+             * {
+             *     "entity": [{
```

Comment:
nitpick: add * to every lines to keep the same style of the comments?

Reply:
added